### PR TITLE
Fetch ads pagination

### PR DESCRIPTION
**Pull request by adeelnoor**
- Updated prisma model of riz_add to have mapped properties
- Added pagination to fetching all ads
- added riz_ad_images table and model

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -8,25 +8,25 @@ datasource db {
...
-  is_featured   Boolean?       @default(true)
-  total_views   Int            @default(autoincrement())
-  is_deleted    Boolean?       @default(false)
-  is_sold       Boolean?       @default(false)
-  currency      String?        @db.VarChar
-  ad_location   riz_locations? @relation(fields: [location_id], references: [id], onDelete: NoAction, onUpdate: NoAction)
-  ad_user       riz_users?     @relation(fields: [user_id], references: [id], onDelete: NoAction, onUpdate: NoAction)
-  ad_tags       riz_ad_tags[]
+  id           Int             @id @default(autoincrement())
+  userId       Int?            @map("user_id")
```
Comment (Mutahir2016) [inline prisma/schema.prisma:12]: 👍

File: Controllers/AdsFetchController.js [hunk 2]
```diff
@@ -1,40 +1,59 @@
...
-          res.status(200).json(mappedAdsWithTagNames);
-    }
-    catch (error) {
-        res.status(500).json({error: error.message});
-    }
+  try {
+    // Pagination defaults with ternary operators
+    const page = parseInt(req.query.page)
+    const limit = parseInt(req.query.limit)
+    const skip = (page - 1) * limit;
```
Comment (Mutahir2016) [inline Controllers/AdsFetchController.js:12]: can't see its usage

File: prisma/schema.prisma [hunk 3]
```diff
@@ -8,25 +8,25 @@ datasource db {
 }
 
 model riz_ad {
-  id            Int            @id @default(autoincrement())
```
Comment (Mutahir2016) [inline prisma/schema.prisma:11]: in addition we can also include the following to help FE understand how this response can be paginated
 total count, current page and total pages

**Review by Mutahir2016 — APPROVED** [review]
Looks good. Pls have a look at the comments.

File: prisma/schema.prisma (hunk 3 above)
Comment (adeelnoor) [reply prisma/schema.prisma:11]: oh yes. Good point I will add in the response
- Current page
- Total pages
- Total count

File: Controllers/AdsFetchController.js (hunk 2 above)
Comment (adeelnoor) [reply Controllers/AdsFetchController.js:12]: Forgot to add. Will add it now

**Comment by adeelnoor** [comment]
Updated all changes 👍

File: prisma/schema.prisma (hunk 3 above)
Comment (adeelnoor) [reply prisma/schema.prisma:11]: I think all we need is totalCount? 
current Page will be locally, starting from page 1 to page += 1 
totalPages also no need?  unless currentPage < total ? 
@Mutahir2016

**Pull request merged**